examples/HelloSO3.cpp

```diff
@@ -0,0 +1,40 @@
+#include "sophus/geometry.hpp"
+#include <iostream>
+
+int main() {
+
+    // The following demonstrates the group multiplication of rotation matrices
+
+    // Create rotation matrices from rotations around the x and y and z axes:
+    Sophus::SO3d R1 = Sophus::SO3d::rotX(M_PI/4);
+    Sophus::SO3d R2 = Sophus::SO3d::rotY(M_PI/6);
+    Sophus::SO3d R3 = Sophus::SO3d::rotZ(-M_PI/3);
+
+    std::cout << "The rotation matrices are" << std::endl;
+    std::cout << "R1:\n" << R1.matrix() << std::endl;
+    std::cout << "R2:\n" << R2.matrix() << std::endl;
+    std::cout << "R3:\n" << R3.matrix() << std::endl;
+    std::cout << "Their product R1*R2*R3:\n" << (R1*R2*R3).matrix() << std::endl;
+
+    // Creating vectors is the same as in Eigen
+    Sophus::Vector3d x;
+    x << 0.0, 0.0, 1.0;
+
+    // Rotation matrices can act on vectors
+    std::cout << std::endl;
+    std::cout << "Rotation matrices can act on vectors" << std::endl;
+    std::cout << "x\n" << x << std::endl;
+    std::cout << "R2*x\n" << R2*x << std::endl;
+    std::cout << "R1*(R2*x)\n" << R1*(R2*x) << std::endl;
+    std::cout << "(R1*R2)*x\n" << (R1*R2)*x << std::endl;
+
+    // You can also use SO(3) objects as unit quaternions.
```

Comment:
I'm not so happy with the quaternion example since what you are doing here is not encouraged. 
I'd probably rather replace this section with the following:

```
// SO(3) are internally represented as unit quaternions.
std::cout << "R1 in matrix form\n" << R1,matrix() << std::endl;
std::cout << "R1 in unit quaternion form: " << R1.unit_quaternion().coeffs() << std::endl;
// Note that the order of coefficiences of Eigen's quaternion class is (imag0, imag1, imag2, real) 
```


See comment below.